pkg/appmgmt/general/podevent_handler.go

```diff
@@ -127,6 +128,13 @@ func (p *PodEventHandler) addPod(pod *v1.Pod, eventSource EventSource) interface
 				Metadata: appMeta,
 			})
 		} else {
+			if conf.GetSchedulerConf().GetAllowSimilarAppIdsByDifferentUsers() && app.GetApplicationState() == "Running" && app.GetUser() != appMeta.User {
```

Comment:
Why does the app need to be in a running state? I would expect all states to match this. Otherwise you can still get passed if an app is waiting to be scheduled.

`GetAllowSimilarAppIdsByDifferentUsers` does not seem to cover what is specified: if it returns true you make check that the users do not match. I would expect the check for the setting to be more like: `GetSingleUserPerApplication()` if true app user and task user must match. If not matched reject, otherwise allow it.
So:
```
if conf.GetSchedulerConf().GetSingleUserPerApplication() && app.GetUser() != appMeta.User {
     reject the task
}
```